docs/source/api/passes.rst

```diff
@@ -174,9 +174,3 @@ VitisAIQuantization
 OptimumConversion
 --------------------
 .. autoconfigclass:: olive.passes.OptimumConversion
-
-.. _optimum_merging:
-
-OptimumMerging
```

Comment:
I don't think we should remove this pass entirely. It can be renamed and use the new merge utils. 

For passes like transformers optimization, quantization, inc quantization, it is not clear if the final model after the following workflows will be the same:
- convert -> optimize each component -> merge
- convert + merge -> optimize merged model 

So I think it is better to keep the option to merge later open.

Reply:
`merge_decoders` logic is the same as what `OptimumMerge` is. Instead of it being a separate pass, it is merely a utility function. If need be we can create pass that just calls the `merge_decoders` function instead of using `optimum`.

Reply:
Yes, that's what I am suggesting too. Until we can confirm that merging first and then running the other passes later will produce the same model as merging later, I don't think we should change the llama or dolly examples to merge first.

Reply:
Yeah I agree with Jambay. We could create a new pass to have an option to merge models later.

Reply:
Puled out the change to OptimumMerge & OptimumModelHandler. PR focus on introducing merge decoders for OnnxConversion only.